### PR TITLE
FIX: show ram calculation error reason to player

### PR DESCRIPTION
**Pull request by bezrodnov**
There was a discussion in Discord where people complained about not being able to understand where exactly syntax error was detected in their code. This PR should give players a bit more transparency on what went wrong.

Now when RAM button is clicked in the Code Editor toolbar player should see where syntax error was found:

image

And the same for running such script in the Terminal:

image

PR is also adding more info in case of import error:

image

image

### CODE REVIEW

**Comment by bezrodnov** [comment]
Without hiding whitespace differences reviewing this is a bit painful

image

File: src/Script/RamCalculations.ts [hunk 1]
```diff
@@ -80,115 +91,108 @@ function parseOnlyRamCalculate(otherScripts: Map<ScriptFilePath, Script>, code:
...
+  // Process additional modules, which occurs if the "main" script has any imports
+  while (parseQueue.length > 0) {
+    const nextModule = parseQueue.shift();
+    if (nextModule === undefined) throw new Error("nextModule should not be undefined");
+    if (nextModule.startsWith("https://") || nextModule.startsWith("http://")) continue;
 
-      resolvedRefs.add(ref);
+    // Using root as the path base right now. Difficult to implement
+    const filename = resolveScriptFilePath(nextModule, root, ns1 ? ".script" : ".js");
+    if (!filename) return { errorCode: RamCalculationErrorCode.ImportError }; // Invalid import path
```
Comment (bezrodnov) [inline src/Script/RamCalculations.ts]: we could actually make this transparent to players too by adding fileName to `error` (and probably changing it to `errorMessage` (string | undefined). wdyt?

Comment (bezrodnov) [reply src/Script/RamCalculations.ts]: I assume it totally makes sense and added it as well.

File: src/Script/Script.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ export class Script implements ContentFile {
   // Ram calculation, only exists after first poll of ram cost after updating
   ramUsage: number | null = null;
   ramUsageEntries: RamUsageEntry[] = [];
+  ramCalculationError?: string | null = null;
```
Comment (d0sboots) [inline src/Script/Script.ts]: This shouldn't be both optional *and* nullable. Nullable is preferred, for monomorphism.

**Review by d0sboots — APPROVED** [review]

**Review by Snarling — COMMENTED** [review]
Seeing console error spam now when editing file (or any other time ram is calculated) (sorry about the weird noise on video)

https://github.com/bitburner-official/bitburner-src/assets/84951833/e8fc39f4-c818-4a90-a2da-29d1bd72cbca

Other than that, looks good.

File: src/Script/Script.ts (hunk 2 above)
Comment (bezrodnov) [reply src/Script/Script.ts]: forgot to remove that

**Pull request merged**